x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecurityIntegTestCase.java

```diff
@@ -380,22 +381,38 @@ protected Function<Client, Client> getClientWrapper() {
         return client -> (client instanceof NodeClient) ? client.filterWithHeader(headers) : client;
     }
 
+    /**
+     * Waits for security index to become available. Note that you must ensure index creation was triggered before calling this method,
+     * either by calling one of the resource creation APIs (e.g., creating a user), or by calling {@link #createSecurityIndex()}.
+     */
     public void assertSecurityIndexActive() throws Exception {
         assertSecurityIndexActive(cluster());
     }
 
+    protected void createSecurityIndex() {
+        final Client client = client().filterWithHeader(
+            Collections.singletonMap(
+                "Authorization",
+                UsernamePasswordToken.basicAuthHeaderValue(
+                    SecuritySettingsSource.ES_TEST_ROOT_USER,
+                    SecuritySettingsSourceField.TEST_PASSWORD_SECURE_STRING
+                )
+            )
+        );
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest(SECURITY_MAIN_ALIAS);
+        client.admin().indices().create(createIndexRequest).actionGet();
+    }
+
     public void assertSecurityIndexActive(TestCluster testCluster) throws Exception {
         for (Client client : testCluster.getClients()) {
             assertBusy(() -> {
                 ClusterState clusterState = client.admin().cluster().prepareState(TEST_REQUEST_TIMEOUT).setLocal(true).get().getState();
                 assertFalse(clusterState.blocks().hasGlobalBlock(GatewayService.STATE_NOT_RECOVERED_BLOCK));
                 Index securityIndex = resolveSecurityIndex(clusterState.metadata());
-                if (securityIndex != null) {
```

Comment:
This is the actual fix: because of the null check, we don't get tripped assertions if the index does exist and assertBusy completes successfully (same for the inner indexRoutingTable null check)